drivers/modem/gsm_ppp.c

```diff
@@ -1058,6 +1060,9 @@ void gsm_ppp_stop(const struct device *dev)
 
 	net_if_l2(iface)->enable(iface, false);
 
+	/* wait for the interface to be properly down */
+	k_sem_take(&gsm->sem_if_down, K_FOREVER);
```

Comment:
Some nits: cast unhandled return to `(void)`

Reply:
See https://github.com/zephyrproject-rtos/zephyr/blob/4bf9388080dc92e5defbfd7074c360321aa69185/drivers/modem/gsm_ppp.c#L150

Reply:
Thanks, applied